File: src/main/kotlin/dclab/powerdatabackend/service/DataService.java
package dclab.powerdatabackend.service;


import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
import dclab.powerdatabackend.mapper.OldDatasMapper;
import dclab.powerdatabackend.model.Datas;
import org.springframework.stereotype.Service;


@Service
public class DataService extends ServiceImpl<OldDatasMapper, Datas> {

}

File: src/main/kotlin/dclab/powerdatabackend/service/DataService.kt
package dclab.powerdatabackend.service

import dclab.powerdatabackend.domain.DataInfo
import dclab.powerdatabackend.mapper.OldDataMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class DataService{
    @Autowired
    val oldDataMapper:OldDataMapper? = null

    fun search(measurePoint:String,location:String,startTime:String?,endTime:String?):Array<DataInfo>{
        return oldDataMapper!!.getSearchArray("%$measurePoint%","%$location%",if(startTime==null)"0" else startTime.replace("-","")
                ,if(endTime==null)"999999" else endTime.replace("-",""))
    }
}